src/server/services/wfs3/qgswfs3handlers.cpp

```diff
@@ -1231,8 +1231,7 @@ void QgsWfs3CollectionsItemsHandler::handleRequest( const QgsServerApiContext &c
 
       if ( ! filterRect.isNull() )
       {
-        const QgsCoordinateTransform ct( bboxCrs, mapLayer->crs(), context.project()->transformContext() );
-        ct.transform( filterRect );
+        const QgsCoordinateTransform ct( bboxCrs, crs, context.project()->transformContext() );
```

Comment:
(unrelated) Doesn't this block a try/catch QgsCsException? Or is that handled at a higher level?

Reply:
Exceptions are actually catched at a higher level but it makes sense to throw a message here, I've added it.